src/Service/BroadcastsService.php

```diff
@@ -51,4 +52,71 @@ public function findByServiceAndDateRange(
 
         return $this->mapManyEntities($dbEntities);
     }
+
+    public function groupBroadcastsByPeriodOfDay(array $broadcasts, DateTimeImmutable $selectedDate)
+    {
+        $intervalsDay = [
+            'early' => [],
+            'morning' => [],
+            'afternoon' => [],
+            'evening' => [],
+            'late' => []
+        ];
+
+        $prior_broadcast = null;
+        foreach($broadcasts as $broadcast) {
+            // // If the end of the prior is earlier than the start of this broadcast
```

Comment:
I know we are not handling gasp broadcast for now in frontend, so not sure what to do regarding to this.